support/hooks/pre-push

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+#
+# An example hook script to verify what is about to be pushed.  Called by "git
+# push" after it has checked the remote status, but before anything has been
+# pushed.  If this script exits with a non-zero status nothing will be pushed.
+#
+# This hook is called with the following parameters:
+#
+# $1 -- Name of the remote to which the push is being done
+# $2 -- URL to which the push is being done
+#
+# If pushing without using a named remote those arguments will be equal.
+#
+# Information about the commits which are being pushed is supplied as lines to
+# the standard input in the form:
+#
+#   <local ref> <local sha1> <remote ref> <remote sha1>
+#
+# This sample shows how to prevent push of commits where the log message starts
+# with "WIP" (work in progress).
+
+DUMMY_SHA=0000000000000000000000000000000000000000
+
+echo "Running pre-push check; to skip this step use 'push --no-verify'"
+
+while read LOCAL_REF LOCAL_SHA REMOTE_REF REMOTE_SHA
+do
+	if [ "$LOCAL_SHA" = $DUMMY_SHA ]
+	then
+		# Branch deleted. Do nothing.
```

Comment:
Can you use two space indents? That's what we've been doing in general across most languages and shell scripts in the Envoy code base. Thanks.